src/coreclr/jit/emit.h

```diff
@@ -2761,6 +2761,14 @@ class emitter
 
     bool emitForceStoreGCState;
 
+    // This flag is used together with `emitForceStoreGCState`. After we set
+    // emitForceStoreGCState = true, we will mark `emitAddedLabel` to true whenever
+    // we see a label IG. In emitSavIG, we will reset `emitForceStoreGCState = false`
+    // only after seeing `emitAddedLabel == true`. Until then, we will keep recording
+    // GC_VARS on the IGs.
+
+    bool emitAddedLabel;
```

Comment:
Maybe for cleanliness you should initialize `emitAddedLabel = false` in `emitter::emitBegFN`, alongside the existing `emitForceStoreGCState = false`